doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -619,32 +619,44 @@ file is a particular case of a module called a *library file*.
 
 .. cmd:: Declare ML Module {+ @string }
 
-   This commands dynamically loads OCaml plugins dynamically.  The string
-   argument must be a valid `findlib <http://projects.camlcity.org/projects/findlib.html>`
-   plugin name, for example ``coq-core.plugins.ltac``. The first component of
-   the plugin name is a package that has to be in scope of ``findlib``.
+   This commands will load an OCaml "plugin" and its dependencies
+   dynamically. The string argument must be a valid `findlib
+   <http://projects.camlcity.org/projects/findlib.html>` library name,
+   for example ``coq-core.plugins.ltac``. The first component of the
+   plugin name is a package that has to be in scope of ``findlib``'s
+   search path.
+
    One can see the paths explored by ``findlib`` by running
    ``ocamlfind printconf`` and get the list of available packages
    by running ``ocamlfind list | grep coq`` (Coq packages are typically named
    ``coq-something``).
 
    This command is reserved for plugin developers, who should provide
-   a .v file containing the command. Users of the plugins will then generally
-   load the .v file.
+   a ``.v`` file containing the command. Users of the plugins will
+   usually require the resulting ``.vo`` file which will then
+   transitively load the required plugin.
+
+   Note that the plugin loading system for Coq changed in 8.16 to use
+   findlib. Previous Coq versions loaded OCaml dynamic objects by
+   first locating the object file from ``-I`` directives, then
+   directly invoking ``Dynlink.loadfile``. For compatibility purposes,
+   8.16 still supports this legacy method, with the syntax being
+   ``Declare ML Module "foo_plugin:pkg.plugin.foo".``, where
+   ``foo_plugin`` is the name of the OCaml object file.
```

Comment:
For clarity, mention this where you first explain the string appearing in the command (line 623) rather then 20 lines later.  Or move it.